### PR TITLE
Update Runtime Weights

### PR DESCRIPTION
**Pull request by liamaharon**
For https://github.com/polkadot-fellows/runtimes/issues/68#issuecomment-1849742955.

Uses instructions introduced to the readme in https://github.com/polkadot-fellows/runtimes/pull/127 to re-bench all runtimes.

TODO
- [x] Is this much change % expected? Is benchmarking on vCPUs flawed because they are arbitrarily shared with other cloud customers? 
  - It seems to be fine, the changes align with actual changes in configuration of the runtimes (thanks @ggwpez!).
- [x] Polkadot
- [x] Kusama
- [x] Asset Hub Polkadot
- [x] Asset Hub Kusama
- [x] Bridge Hub Polkadot
- [x] Bridge Hub Kusama
- [x] Collectives Polkadot
- [ ] Look into why subweight shows some errors

## Polkadot Changes

Screenshot 2023-12-19 at 11 06 20
Screenshot 2023-12-19 at 11 08 24
Screenshot 2023-12-19 at 11 10 14

## Kusama Changes

Screenshot 2023-12-19 at 11 12 19
Screenshot 2023-12-19 at 11 12 43

## Asset Hub Polkadot Changes

Screenshot 2023-12-19 at 20 34 25

## Asset Hub Kusama Changes

Screenshot 2023-12-19 at 20 43 37

## Bridge Hub Polkadot

Screenshot 2023-12-20 at 11 49 13

## Bridge Hub Kusama

Screenshot 2023-12-20 at 11 48 39

## Collectives Polkadot

Screenshot 2023-12-20 at 11 50 33

Screenshot 2023-12-20 at 11 51 07

### CODE REVIEW

File: relay/polkadot/src/weights/pallet_preimage.rs [hunk 1]
```diff
@@ -45,190 +51,219 @@ use core::marker::PhantomData;
...
 		//  Estimated: `3556`
-		// Minimum execution time: 50_195_000 picoseconds.
-		Weight::from_parts(55_889_000, 0)
+		// Minimum execution time: 55_238_000 picoseconds.
+		Weight::from_parts(64_350_000, 0)
 			.saturating_add(Weight::from_parts(0, 3556))
-			.saturating_add(T::DbWeight::get().reads(1))
-			.saturating_add(T::DbWeight::get().writes(2))
+			.saturating_add(T::DbWeight::get().reads(3))
+			.saturating_add(T::DbWeight::get().writes(3))
```
Comment (ggwpez) [inline relay/polkadot/src/weights/pallet_preimage.rs:129]: Preimage reads/writes increased. This could be why its slower now.

**Comment by bkontur** [comment]
it would be cool to include also this cleaning PR: https://github.com/polkadot-fellows/runtimes/pull/57

**Comment by liamaharon** [comment]
Weights are all updated now. One small thing I'm not sure of is why `subweight` is printing errors for extrinsics which seem to have been updated without issue, e.g.

Screenshot 2023-12-20 at 11 56 04

Screenshot 2023-12-20 at 11 58 54

@ggwpez any ideas?

**Review by bkchr — APPROVED** [review]

File: relay/kusama/src/lib.rs [hunk 2]
```diff
@@ -220,7 +220,7 @@ pub struct OriginPrivilegeCmp;
 impl PrivilegeCmp<OriginCaller> for OriginPrivilegeCmp {
 	fn cmp_privilege(left: &OriginCaller, right: &OriginCaller) -> Option<Ordering> {
 		if left == right {
-			return Some(Ordering::Equal)
+			return Some(Ordering::Equal);
```
Comment (joepetrowski) [inline relay/kusama/src/lib.rs:223]:
```suggestion
			return Some(Ordering::Equal)
```

File: relay/kusama/src/lib.rs [hunk 3]
```diff
@@ -2598,7 +2598,7 @@ mod remote_tests {
 	#[tokio::test]
 	async fn run_migrations() {
 		if var("RUN_MIGRATION_TESTS").is_err() {
-			return
+			return;
```
Comment (joepetrowski) [inline relay/kusama/src/lib.rs:2601]:
```suggestion
			return
```

File: relay/kusama/src/lib.rs [hunk 4]
```diff
@@ -2675,12 +2675,12 @@ mod init_state_migration {
 		fn on_runtime_upgrade() -> frame_support::weights::Weight {
 			if AutoLimits::<Runtime>::get().is_some() {
 				log::warn!("Automatic trie migration already started, not proceeding.");
-				return <Runtime as frame_system::Config>::DbWeight::get().reads(1)
+				return <Runtime as frame_system::Config>::DbWeight::get().reads(1);
```
Comment (joepetrowski) [inline relay/kusama/src/lib.rs:2678]:
```suggestion
				return <Runtime as frame_system::Config>::DbWeight::get().reads(1)
```

File: relay/kusama/src/lib.rs [hunk 5]
```diff
@@ -2675,12 +2675,12 @@ mod init_state_migration {
...
 			};
 
 			if MigrationProcess::<Runtime>::get() != Default::default() {
 				log::warn!("MigrationProcess is not Default. Not proceeding.");
-				return <Runtime as frame_system::Config>::DbWeight::get().reads(2)
+				return <Runtime as frame_system::Config>::DbWeight::get().reads(2);
```
Comment (joepetrowski) [inline relay/kusama/src/lib.rs:2683]:
```suggestion
				return <Runtime as frame_system::Config>::DbWeight::get().reads(2)
```

**Review by joepetrowski — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

**Review by acatangiu — APPROVED** [review]

**Comment by ggwpez** [comment]
The two remaining failing tests are from https://github.com/polkadot-fellows/runtimes/pull/114 and could be commented for now.

**Review by ordian — APPROVED** [review]

**Comment by acatangiu** [comment]
> The two remaining failing tests are from #114 and could be commented for now.

I'm looking into this now, so far looks like actual weights issue with `Error(WeightNotComputable)`:

> AssetHubPolkadot TRACE xcm::weight: WeightInfoBounds message: Xcm([ReserveAssetDeposited(MultiAssets([MultiAsset { id: Concrete(MultiLocation { parents: 1, interior: Here }), fun: Fungible(10000000000000) }])), ClearOrigin, BuyExecution { fees: MultiAsset { id: Concrete(MultiLocation { parents: 1, interior: Here }), fun: Fungible(10000000000000) }, weight_limit: Unlimited }, DepositAsset { assets: Wild(AllCounted(1)), beneficiary: MultiLocation { parents: 0, interior: X1(AccountId32 { network: None, id: [142, 175, 4, 21, 22, 135, 115, 99, 38, 201, 254, 161, 126, 37, 252, 82, 135, 97, 54, 147, 201, 18, 144, 156, 178, 38, 170, 71, 148, 242, 106, 72] }) } }, SetTopic([201, 32, 171, 168, 92, 206, 203, 109, 135, 135, 85, 253, 130, 152, 157, 65, 206, 56, 147, 183, 121, 12, 173, 10, 97, 204, 25, 152, 187, 87, 194, 148])])    

> AssetHubPolkadot events: [RuntimeEvent::DmpQueue(Event::ExecutedDownward { message_hash: [39, 4, 119, 101, 74, 2, 62, 188, 202, 127, 77, 91, 0, 138, 171, 118, 232, 89, 9, 132, 77, 17, 6, 77, 108, 102, 102, 30, 56, 102, 16, 78], message_id: [39, 4, 119, 101, 74, 2, 62, 188, 202, 127, 77, 91, 0, 138, 171, 118, 232, 89, 9, 132, 77, 17, 6, 77, 108, 102, 102, 30, 56, 102, 16, 78], outcome: Error(WeightNotComputable) })]

**Comment by bkontur** [comment]
> ReserveAssetDeposited

AssetHubPolkadot/Kusama does not recognize any reserve locations `IsReserve = ()` [here](https://github.com/liamaharon/runtimes/blob/liam-update-weights/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L440-L443), 
`tests::reserve_transfer::limited_reserve_transfer_native_asset_from_relay_to_system_para_fails` fails which is ok, but we need to just adjust assert [here](https://github.com/liamaharon/runtimes/blob/liam-update-weights/integration-tests/emulated/assets/asset-hub-polkadot/src/tests/reserve_transfer.rs#L38C1-L40C2)

File: system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs [hunk 6]
```diff
@@ -1,190 +1,208 @@
...
+	/// Storage: `Benchmark::Override` (r:0 w:0)
+	/// Proof: `Benchmark::Override` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	pub(crate) fn reserve_asset_deposited() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 500_000_000_000 picoseconds.
-		Weight::from_parts(500_000_000_000, 0)
+		// Minimum execution time: 18_446_744_073_709_551_000 picoseconds.
+		Weight::from_parts(18_446_744_073_709_551_000, 0)
```
Comment (acatangiu) [inline system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs:107]: This is now incorrectly set to `Weight::MAX` - looks like the issue is with benchmarks - I can look into it later today unless somebody knows off the top of their heads what the issue is

**Comment by acatangiu** [comment]
> > ReserveAssetDeposited
> 
> AssetHubPolkadot/Kusama does not recognize any reserve locations `IsReserve = ()` [here](https://github.com/liamaharon/runtimes/blob/liam-update-weights/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L440-L443), `tests::reserve_transfer::limited_reserve_transfer_native_asset_from_relay_to_system_para_fails` fails which is ok, but we need to just adjust assert [here](https://github.com/liamaharon/runtimes/blob/liam-update-weights/integration-tests/emulated/assets/asset-hub-polkadot/src/tests/reserve_transfer.rs#L38C1-L40C2)

no, the assert is correct - it works fine without the bad weights - the problem is certain XCMs are unweighable with this PR because multiple XCM instructions have been benchmarked to `Weight::MAX`

this is a valid weights issue, not a test issue

**Comment by bkontur** [comment]
> > > ReserveAssetDeposited
> > 
> > 
> > AssetHubPolkadot/Kusama does not recognize any reserve locations `IsReserve = ()` [here](https://github.com/liamaharon/runtimes/blob/liam-update-weights/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L440-L443), `tests::reserve_transfer::limited_reserve_transfer_native_asset_from_relay_to_system_para_fails` fails which is ok, but we need to just adjust assert [here](https://github.com/liamaharon/runtimes/blob/liam-update-weights/integration-tests/emulated/assets/asset-hub-polkadot/src/tests/reserve_transfer.rs#L38C1-L40C2)
> 
> no, the assert is correct - it works fine without the bad weights - the problem is certain XCMs are unweighable with this PR because multiple XCM instructions have been benchmarked to `Weight::MAX`
> 
> this is a valid weights issue, not a test issue


That previous value `500_000_000_000` (`T::BlockWeights::get().max_block`) was kind of hack as a part  of remote weight estimation which was removed.

This `Weight::MAX` is correct expected value for `ReserveAssetDeposited` instruction which is not supported on AssetHub because of `IsReserve = ()`, so benchmark and weight is ok. And `XcmExecutor` really reports `WeightNotComputable` for `Weight::MAX`.

https://github.com/paritytech/polkadot-sdk/pull/1726

**Comment by bkontur** [comment]
> > > > ReserveAssetDeposited
> > > 
> > > 
> > > AssetHubPolkadot/Kusama does not recognize any reserve locations `IsReserve = ()` [here](https://github.com/liamaharon/runtimes/blob/liam-update-weights/system-parachains/asset-hubs/asset-hub-polkadot/src/xcm_config.rs#L440-L443), `tests::reserve_transfer::limited_reserve_transfer_native_asset_from_relay_to_system_para_fails` fails which is ok, but we need to just adjust assert [here](https://github.com/liamaharon/runtimes/blob/liam-update-weights/integration-tests/emulated/assets/asset-hub-polkadot/src/tests/reserve_transfer.rs#L38C1-L40C2)
> > 
> > 
> > no, the assert is correct - it works fine without the bad weights - the problem is certain XCMs are unweighable with this PR because multiple XCM instructions have been benchmarked to `Weight::MAX`
> > this is a valid weights issue, not a test issue
> 
> That previous value `500_000_000_000` (`T::BlockWeights::get().max_block`) was kind of hack as a part of remote weight estimation which was removed.
> 
> This `Weight::MAX` is correct expected value for `ReserveAssetDeposited` instruction which is not supported on AssetHub because of `IsReserve = ()`, so benchmark and weight is ok. And `XcmExecutor` really reports `WeightNotComputable` for `Weight::MAX`.
> 
> [paritytech/polkadot-sdk#1726](https://github.com/paritytech/polkadot-sdk/pull/1726)

My point is that`ReserveAssetDeposited` is not supported on AssetHub (`IsReserve = ()`), so the benchmark weight is unknown aka `Weight::MAX` (because we don't know real implementation - how many reads...), so the `XcmExecutor` does not even try to execute it.
Previously with `T::BlockWeights::get().max_block = 500_000_000_000`, it tried to execute it (is it ok?) but with `UntrustedReserveLocation` failure from processing of `ReserveAssetDeposited`.

I changed that `T::BlockWeights::get().max_block` to `Weight::MAX` according to the discussion here: https://github.com/paritytech/cumulus/pull/2934#discussion_r1273439749

Maybe I am wronng, but I think it is safer not to execute at all, if we don't know the real benchmark weight beforehand.

**Comment by acatangiu** [comment]
```rust
		let (trusted_reserve, transferable_reserve_asset) = T::TrustedReserve::get()
			.ok_or(BenchmarkError::Override(
				BenchmarkResult::from_weight(T::BlockWeights::get().max_block)
				BenchmarkResult::from_weight(Weight::MAX)
			))?;
```

Ok, so runtimes which don't recognize any reserve locations, will fail on initial XCM weighing instead of failing on barrier a bit later. That's fine, and I agree it's better to use Weight::MAX instead of some dummy value.
Expectation in this case is that if we ever add trusted locations we also need to change benchmarking code - which again is fine and this test will actually force us to. 👍 

Changing test to expect weight failure instead of barrier failure.

**Comment by acatangiu** [comment]
@liamaharon I'm not allowed to push into your fork:

```
kata@MacBook-Pro runtimes % git push
ERROR: Permission to liamaharon/runtimes.git denied to acatangiu.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Please add the fix yourself, you can find it here: https://github.com/acatangiu/runtimes/commit/48787a1b3fbbe55b1ff9074284a72923d10acc76

**Comment by ggwpez** [comment]
@liamaharon is OOO but AFAIK i can push 🤷   
Will run the test locally and then push.

**Pull request merged**